### ISSUE TITLE
Pointers to functions

### ISSUE THREAD

**Issue by Qpupier** [42school/norminette#381]
Hello,

Norminette only allowed pointer functions in typedef
A fix will be deployed to accept all definitions of pointer functions ?

Thanks

**Comment by yumamur** [comment]
$> cat main.c && norminette main.c
1 #include <fcntl.h>
2 #include <unistd.h>
3
4 typedef void    (*t_funccast)(int, void *);
5
6 void    function(int fd, char *str)
7 {
8         write(fd, str, 3);
9 }
10
11 void    file_f(char *fname, void (*func)(int, void *), void *param)
12 {
13         int     fd;
14
15         fd = open(fname, O_WRONLY);
16         if (fd == -1)
17                 return ;
18         func(fd, param);
19         close(fd);
20 }
21
22 int     main(int argc, char *argv[])
23 {
24         file_f(argv[argc - 1], (void (*)(int , void *))function, "foo");
25         file_f(argv[argc - 1], (void (*)(int, void *))function, "foo");
26         file_f(argv[argc - 1], (t_funccast)function, "foo");
27 }
main.c: Error!
Error: INVALID_HEADER       (line:   1, col:   1):      Missing or invalid 42 header
Error: NO_SPC_BFR_OPR       (line:  24, col:  42):      extra space before operator
Error: SPACE_AFTER_KW       (line:  25, col:  38):      Missing space after keyword

**Comment by wil-pedago** [comment]
Yes, and it is done. Function pointer declarations outside a typedef are accepted on 3.3.60, including a function returning one:

```c
int	(*g_get(void))(int);
```

Reported `OK!`. See #506 and #528, both fixed in #595.
